widget/editor.go

```diff
@@ -936,6 +1153,115 @@ func (e *Editor) NumLines() int {
 	return len(e.lines)
 }
 
+// If there's a selection, move to its earliest point and delete the selected
+// text, leaving the editor caret at that point, and clearing the selection.
+//
+// If there's not a selection, don't do anything.
+func (e *Editor) DeleteSelection() {
+	l := e.Selection.Len()
+	if l == 0 {
+		return
+	}
+	e.rr.caret = min(e.Selection.Start.ofs, e.Selection.End.ofs)
+	e.ClearSelection()
+	e.Delete(l)
+}
+
+func (e *Editor) ClearSelection() {
```

Comment:
I suggest unexporting this method; it seems to be a side-effect of higher-level operations (move, delete etc.)

Reply:
> it seems to be a side-effect of higher-level operations 

It is within the Editor code, but I also wanted there to be an easy, obvious way to clear the selection programmatically.  I guess since `Editor.Selection` and `type Selection` are both exported, a user could just say `e.Selection = Selection{}` directly.

Actually though, probably _neither_ should be exported, unless I document that it shouldn't be written to.  Setting up a Selection correctly is tricky, and probably should only be done by setting `e.Selection.Start.ofs` and `e.Selection.End.ofs` and calling `Selection.makeValid` (just as I do now for the "select all" (`Shortcut-A`) code).

How about I unexport both, and add these:
* `func (e *Editor) Selection() (startOfs, endOfs int)` to get the selection offsets
* `func (e *Editor) SetSelection(startOfs, endOfs int)` to set the selection
* `func (e *Editor) SelectedText() string` to get the selected text.  Or should this just be another return value of `Selction()`, above?

Reply:
The three replacement methods SGTM.